From 5044cfd2a0d615ed1b1b3c72de001ec64483446c Mon Sep 17 00:00:00 2001
From: Michael Edwards <michaedw@cisco.com>
Date: Tue, 13 Jul 2010 15:14:11 -0700
Subject: [PATCH] Add linux-elf-Os configuration

---
 Configure |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/Configure b/Configure
index e3b13b9..f607e14 100755
--- a/Configure
+++ b/Configure
@@ -344,6 +344,7 @@ my %table=(
 #### IA-32 targets...
 "linux-ia32-icc",	"icc:-DL_ENDIAN -DTERMIO -O2 -no_cpprt::-D_REENTRANT::-ldl:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:linux-shared:-KPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 "linux-elf",	"gcc:-DL_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -Wall::-D_REENTRANT::-ldl:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
+"linux-elf-Os",	"gcc:-DPURIFY -DL_ENDIAN -DTERMIO -g -Os -fomit-frame-pointer -Wall::-D_REENTRANT::-ldl:BN_LLONG DES_PTR DES_RISC1 ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 "linux-aout",	"gcc:-DL_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -march=i486 -Wall::(unknown):::BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_asm}:a.out",
 ####
 "linux-generic64","gcc:-DTERMIO -O3 -Wall::-D_REENTRANT::-ldl:SIXTY_FOUR_BIT_LONG RC4_CHAR RC4_CHUNK DES_INT DES_UNROLL BF_PTR:${no_asm}:dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
-- 
1.7.0

