From fc37f23b9317c80341d9393b8f9f989bff95b5b7 Mon Sep 17 00:00:00 2001
From: Michael Edwards <michaedw@cisco.com>
Date: Sun, 18 Jul 2010 07:43:00 -0700
Subject: [PATCH 1/4] Adjust Makefile for libXmlRpc

---
 Makefile |   15 +++++++--------
 1 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index 9141f81..15c68d5 100644
--- a/Makefile
+++ b/Makefile
@@ -4,8 +4,8 @@ SRC		= ./src
 SHARED		= -shared
 CPPFLAGS	= -I$(SRC)
 DEBUG		= -g
-OPTIMIZE	= -O2
-GCCWARN		= -Wall -Wstrict-prototypes
+OPTIMIZE	= -Os -DUSE_OPENSSL
+GCCWARN		= -Wall -W
 CXXFLAGS	= $(DEBUG) $(GCCWARN) $(OPTIMIZE) $(INCLUDES)
 VERSION		= 0.8
 
@@ -16,8 +16,7 @@ CP		= cp
 LN		= ln -s
 
 LIB		= ./libXmlRpc.a
-LIBALT		= ./libxmlrpc++.a
-SO		= ./libxmlrpc++.so.$(VERSION)
+SO		= ./libXmlRpc.so.$(VERSION)
 
 # Add your system-dependent network libs here. These are
 # only used to build the tests (your application will need them too).
@@ -36,10 +35,10 @@ all:		$(LIB) $(SO) tests
 
 $(LIB):		$(OBJ)
 		$(AR) $(ARFLAGS) $(LIB) $(OBJ)
-		cp $(LIB) $(LIBALT)
 
 $(SO):		$(OBJ)
 		$(CXX) -o $(SO) $(SHARED) $(OBJ)
+		$(LN) $(SO) libXmlRpc.so
 
 
 tests:		$(LIB)
@@ -53,7 +52,7 @@ distclean: clean
 clean:
 		rm -f $(SRC)/*.o
 		rm -f $(SRC)/*~
-		rm -f $(LIB) $(LIBALT) $(SO)
+		rm -f $(LIB) $(SO)
 		rm -f build
 		cd test && $(MAKE) clean
 
@@ -62,8 +61,8 @@ install:
 		$(CP) $(SO) $(LIB) $(DESTDIR)$(prefix)/lib
 		$(MKDIR) $(DESTDIR)$(prefix)/include
 		$(CP) src/*.h $(DESTDIR)$(prefix)/include
-		( cd $(DESTDIR)$(prefix)/lib; rm -f libxmlrpc++.so; $(LN) $(SO) libxmlrpc++.so )
-		# Does not install tests right now	
+		( cd $(DESTDIR)$(prefix)/lib; rm -f libXmlRpc.so; $(LN) $(SO) libXmlRpc.so )
+		# Does not install tests right now
 
 # Debian package make target, in case you want to manually build a package
 # from the distribution.
-- 
1.7.0

