Index: pulseaudio-0.9.23/src/modules/alsa/alsa-sink.c
===================================================================
--- pulseaudio-0.9.23.orig/src/modules/alsa/alsa-sink.c	2011-06-28 19:51:49.000000000 -0600
+++ pulseaudio-0.9.23/src/modules/alsa/alsa-sink.c	2011-06-28 19:52:06.000000000 -0600
@@ -476,7 +476,7 @@
 }
 
 static int mmap_write(struct userdata *u, pa_usec_t *sleep_usec, pa_bool_t polled, pa_bool_t on_timeout) {
-    pa_bool_t work_done = TRUE;
+    pa_bool_t work_done = FALSE;
     pa_usec_t max_sleep_usec = 0, process_usec = 0;
     size_t left_to_play;
     unsigned j = 0;
