--- a/gcc/reload1.c	2011-05-06 11:28:27 +0000
+++ b/gcc/reload1.c	2011-06-30 18:42:13 +0000
@@ -6464,11 +6464,25 @@
 		  if (byte == 0)
 		    need_mode = mode;
 		  else
-		    need_mode
-		      = smallest_mode_for_size
-		        (GET_MODE_BITSIZE (mode) + byte * BITS_PER_UNIT,
-			 GET_MODE_CLASS (mode) == MODE_PARTIAL_INT
-			 ? MODE_INT : GET_MODE_CLASS (mode));
+		    {
+		      unsigned int msize
+			= GET_MODE_BITSIZE (mode) + byte * BITS_PER_UNIT;
+		      enum mode_class mclass
+			= (GET_MODE_CLASS (mode) == MODE_PARTIAL_INT)
+			  ? MODE_INT : GET_MODE_CLASS (mode);
+		      if (mclass != MODE_INT)
+			{
+			  enum machine_mode tmpmode;
+			  for (tmpmode = GET_CLASS_NARROWEST_MODE (mclass);
+			       tmpmode != VOIDmode;
+			       tmpmode = GET_MODE_WIDER_MODE (tmpmode))
+			    if (GET_MODE_PRECISION (mode) >= msize)
+			      break;
+			  if (tmpmode == VOIDmode)
+			    mclass = MODE_INT;
+			}
+		      need_mode = smallest_mode_for_size (msize, mclass);
+		    }
 
 		  if ((GET_MODE_SIZE (GET_MODE (last_reg))
 		       >= GET_MODE_SIZE (need_mode))

