From 93b94cfb86186d4f96d7d7ba72735f288a25fee4 Mon Sep 17 00:00:00 2001
From: michaedw in build chroot <build@ctbu-bld4.cisco.com>
Date: Wed, 13 Jul 2011 16:43:49 +0000
Subject: [PATCH] disable doc installation

---
 Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index bce2ad4..aa63fde 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,7 +4,7 @@
 #AUTOMAKE_OPTIONS = foreign 1.6 dist-zip
 AUTOMAKE_OPTIONS = foreign 1.11 dist-zip dist-xz
 
-SUBDIRS = src include doc
+SUBDIRS = src include
 
 m4datadir = $(datadir)/aclocal
 m4data_DATA = ogg.m4
-- 
1.7.0.4

